### DISCUSSION TITLE
Are there plans for a config and storage folder in the user's home directory

### DISCUSSION THREAD

**Discussion by valorin** [laravel-zero/laravel-zero#411] [Ideas]
Are there plans to add support for storing the `.env` and storage folder (i.e. cache, logs, etc) within the users global home directory?
Something like `~/.config/<app-name>`?

I've hacked together a very basic method of putting config in a global folder, and have plans to expand it, but if you're working on a better solution that's even better. Or if it's something you're open to, but haven't done, I can look at building what I need into Zero and submitting a PR?

This is my current very-hacky solution for a cache dir:

```
protected function registerStorage()
{
    $dir = getenv('HOME').'/.config/spirit';

    if (! is_dir($dir)) {
        throw_unless(mkdir($dir, recursive: true), "Unable to create storage directory: '{$dir}'.");
    }

    $this->app->bind('storage-dir', $dir);
    $this->app->get('config')->set('cache.stores.file.path', "{$dir}/cache");
}
```

**Comment by owenvoke** [comment]
Hey @valorin, I've done this sort of thing before using my [Laravel XDG](https://github.com/owenvoke/laravel-xdg) package, as well as [working on a `homeDirectory()` PR](https://github.com/laravel-zero/framework/compare/master...owenvoke:feature/home-directory) at one point. I originally thought it would be better as a package rather than part of Laravel Zero core.

However, I could see a lot of people using it. 👍🏻 So feel free to open a PR with some sort of configuration directory functionality. My only issue before was people like having different naming, locations, etc. for their configuration directories. And also, whether it would be an env file, a PHP config, or what that gets stored.
